lib/Haskell/Prim/Thunk.agda

```diff
@@ -6,10 +6,10 @@ open import Agda.Builtin.Size public
 
 open import Haskell.Prim
 
-record Thunk {ℓ} (a : @0 Size → Set ℓ) (@0 i : Size) : Set ℓ where
+record Thunk {ℓ} (a : {j : Size} → Set ℓ) (i : Size) : Set ℓ where
```

Comment:
No need to bind this `j`, right?

Reply:
It's nice to give it a name so you can pass the argument in a named way (e.g. write `a {j = j}` instead of `a {j}`).